.github/workflows/pull-request.yml

```diff
@@ -5,8 +5,9 @@ on:
       - master
 jobs:
   preview:
-    # Only run this job for events that originate on this repository.
-    if: github.event.pull_request.head.repo.full_name == github.repository
+    # Only run this job for events that originate on this repository and if the automation/merge label is not applied to the PR.
+    # PRs with the automation/tfgen-provider-docs label contain only metadata file changes and don't need a PR build.
+    if: github.event.pull_request.head.repo.full_name == github.repository && !contains(github.event.pull_request.labels, "automation/tfgen-provider-docs")
```

Comment:
I added this part to skip PR builds for those PRs that get opened through automation which contain only metadata file changes. We auto-merge those and don't really need the PR preview links for those since we don't typically pay attention to those.